src/hmacKey.ts

```diff
@@ -0,0 +1,355 @@
+   *
+   * Note: this only create a local reference to an HMAC key, to create
+   * an HMAC key, use {@link Storage#createHmacKey}.
+   *
+   * @param {Storage} storage The Storage instance this HMAC key is
+   *     attached to.
+   * @param {string} accessId The unique accessId for this HMAC key.
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const hmacKey = storage.hmacKey('access-id');
+   */
+  constructor(storage: Storage, accessId: string) {
+    if (!accessId) {
+      throw new Error('An access ID is needed to create an HmacKey object.');
+    }
+
+    const methods = {
+      /**
+       * @typedef {object} DeleteHmacKeyOptions
+       * @property {string} [userProject] This parameter is currently ignored.
+       */
+      /**
+       * @typedef {array} DeleteHmacKeyResponse
+       * @property {object} 0 The full API response.
+       */
+      /**
+       * @callback DeleteHmacKeyCallback
+       * @param {?Error} err Request error, if any.
+       * @param {object} apiResponse The full API response.
+       */
+      /**
+       * Deletes an HMAC key.
+       * Key state must be set to `INACTIVE` prior to deletion.
+       * Caution: HMAC keys cannot be recovered once you delete them.
+       *
+       * The authenticated user must have `storage.hmacKeys.delete` permission for the project in which the key exists.
+       *
+       * @method HmacKey#delete
+       * @param {DeleteHmacKeyOptions} [options] Configuration options.
+       * @param {DeleteHmacKeyCallback} [callback] Callback function.
+       * @returns {Promise<DeleteHmacKeyResponse>}
+       *
+       * @example
+       * const {Storage} = require('@google-cloud/storage');
+       * const storage = new Storage();
+       *
+       * //-
+       * // Delete HMAC key after making the key inactive.
+       * //-
+       * const hmacKey = storage.hmacKey('ACCESS_ID');
+       * hmacKey.update({state: 'INACTIVE'}, (err, hmacKeyMetadata) => {
+       *     if (err) {
+       *       // The request was an error.
+       *       console.error(err);
+       *       return;
+       *     }
+       *     hmacKey.delete((err) => {
+       *       if (err) {
+       *         console.error(err);
+       *         return;
+       *       }
+       *       // The HMAC key is deleted.
+       *     })
+       *   });
+       *
+       * //-
+       * // If the callback is omitted, a promise is returned.
+       * //-
+       * const hmacKey = storage.hmacKey('ACCESS_ID');
+       * hmacKey
+       *   .update({state: 'INACTIVE'})
+       *   .then(() => {
+       *     return hmacKey.delete();
+       *   });
+       */
+      delete: true,
+    };
+
+    super({
+      parent: storage,
+      baseUrl: `/projects/${storage.projectId}/hmacKeys`,
+      id: accessId,
+      methods,
+    });
+
+    this.accessId = accessId;
+    this.metadata = {accessId};
+    this.parent = storage;
```

Comment:
This should be done automatically by the ServiceObject class.

Reply:
fixed.